### ISSUE TITLE
_prefetch doesn't return results

### ISSUE THREAD

**Issue by kevart** [coleifer/peewee#1077]
I have upgraded from peewee==2.6.4 to 2.8.3 and got some issues with _prefetch_ now.

```
prefetched_users = prefetch(db_users, db_emails, db_interactions)

for user in prefetched_users:
    print user.uuid
    '9533af75-a176-40eb-9b24-cdd2dc3e7e5d'
    print user.emails[0].email
    <email address>

    print user.uuid
    '9533af75-a176-40eb-9b24-cdd2dc3e7e5d'
    print user.emails_prefetch[0].email
    IndexError: list index out of range
```

Some more debugging:

```
prefetched_users = prefetch(db_users, db_emails, db_interactions)

print 'users', db_users.count()
50

print 'emails', db_emails.count()
182

    counter = 0
    for user in db_users:

        found = False

        for email in db_emails:
            if email.user_id == user.uuid:
                found = True
                break

        if found:
            print counter, user.uuid, email.email
        else:
            print 'NOT FOUND!'

        counter += 1
```

Prints 50 uuids + emails but no NOT FOUND.

**peewee==2.8.3**

Let me know if you need any other info.

**Comment by kevart** [comment]
Might be on our end. Will collect more info first.

**Comment by kevart** [comment]
Had another query where _prefetch_ didn't return expected results as before.

**Comment by coleifer** [comment]
Can you give me something reproducible?

**Comment by kevart** [comment]
I don't assume you need anything else from me after @alexlatchford's comment.
Do you think you'll get to this soon?

Thanks heaps! ;-)

**Comment by coleifer** [comment]
Should be fixed in 1c237421eaf6d1ab66218e999baa41c1811c5fc2

**Comment by alexlatchford** [comment]
Brilliant thanks Charles!
